lib/tasks/data_hygiene.rake

```diff
@@ -51,42 +51,65 @@ task sync_govdelivery_topic_mappings: :environment do
 
   puts "Fetching topics.."
   topics = Services.gov_delivery.fetch_topics["topics"]
+  topics = topics.map { |topic| [topic["name"], topic["code"]] }
+  subscriber_lists = SubscriberList.where.not(gov_delivery_id: nil)
+                                   .pluck(:title, :gov_delivery_id)
+                                   .map { |title, gov_delivery_id| [(title || gov_delivery_id).strip, gov_delivery_id] }
 
-  if topics.blank?
-    puts "No topics found in GovDelivery."
+  matching = topics & subscriber_lists
+  to_be_deleted = topics - subscriber_lists
+  to_be_created = subscriber_lists - topics
+
+  if to_be_deleted.blank?
+    puts "No topics to be deleted in GovDelivery."
   else
-    puts "Deleting all remote topics.."
-    topics.each do |topic|
-      puts "-- Deleting #{topic["name"]} (#{topic["code"]})"
-      Services.gov_delivery.delete_topic(topic["code"])
+    puts "Deleting remote topics.."
+    threads = to_be_deleted.each_slice(DataHygiene::BATCH_SIZE).map do |batch|
+      Thread.new do
+        batch.each do |name, code|
+          puts "-- Deleting #{name} (#{code})"
+          Services.gov_delivery.delete_topic(code)
+        end
+      end
     end
+    threads.each(&:join)
 
     # GovDelivery delete topics asynchronously and/or are only eventually
     # consistent on deletes.  We have to wait until all the deletes take effect
     # or we'll get conflicts when trying to recreate them.
-    attempts = 0
-    while Array(Services.gov_delivery.fetch_topics["topics"]).count > 0
+    puts 'Wainting for 30 seconds for topics to be asynchronously deleted'
+    30.times do
       sleep 1
-
-      if attempts >= 15
-        puts "Attempted to delete all topics and it doesn't seem to have worked."
-        puts "Trying again may work."
-        abort
-      else
-        attempts += 1
-      end
+      print '.'
     end
   end
 
-  puts "Creating remote topics to match the #{SubscriberList.count} local topics.."
-  created = {}
-  SubscriberList.find_each do |list|
-    next if created.has_key?(list.gov_delivery_id)
+  3.times.each do |i|
```

Comment:
We should check whether there are any topics to be created before going into this loop.